src/reducers/quick-open.js

```diff
@@ -72,3 +74,19 @@ export function getQuickOpenQuery(state: OuterState) {
 export function getQuickOpenType(state: OuterState): QuickOpenType {
   return state.quickOpen.get("searchType");
 }
+export const getFormattedSources = createSelector(getSources, sources =>
+  formatSources(sources)
+);
+
+export const getFormattedSourcesWithTabExistence = createSelector(
+  getFormattedSources,
+  getTabs,
+  (formattedSources, tabs) => {
+    const tabsAsArray = tabs.toArray();
+    const formattedSourcesWithTabExistence = formattedSources.map(source => {
+      source.tabExists = tabsAsArray.includes(source.url);
+      return source;
+    });
+    return formattedSourcesWithTabExistence;
+  }
+);
```

Comment:
this is the first `sources` related selector in quick-open, it is also the first time a reducer has imported ../selectors, which makes me nervous of circular dependencies.

Perhaps, you were right the first time to `source.tabExists = tabsAsArray.includes(source.url);` in formatSources :wink: